client/me/application-passwords/index.jsx

```diff
@@ -1,230 +1,107 @@
-import FormSectionHeading from 'calypso/components/forms/form-section-heading';
-import FormTextInput from 'calypso/components/forms/form-text-input';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import SectionHeader from 'calypso/components/section-header';
-import AppPasswordItem from 'calypso/me/application-password-item';
+import useAppPasswordsQuery from 'calypso/data/application-passwords/use-app-passwords-query';
+import useCreateAppPasswordMutation from 'calypso/data/application-passwords/use-create-app-password-mutation';
 import { recordGoogleEvent } from 'calypso/state/analytics/actions';
-import {
-	clearNewApplicationPassword,
-	createApplicationPassword,
-} from 'calypso/state/application-passwords/actions';
-import getApplicationPasswords from 'calypso/state/selectors/get-application-passwords';
-import getNewApplicationPassword from 'calypso/state/selectors/get-new-application-password';
-
+import { errorNotice } from 'calypso/state/notices/actions';
+import NewAppPasswordForm from './form';
+import AppPasswordsList from './list.jsx';
+import NewAppPassword from './new-password';
 import './style.scss';
-class ApplicationPasswords extends Component {
-	static initialState = Object.freeze( {
-		applicationName: '',
-		addingPassword: false,
-		submittingForm: false,
-	} );
-
-	state = this.constructor.initialState;
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.state.submittingForm && ! this.props.newAppPassword && !! nextProps.newAppPassword ) {
-			this.setState( { submittingForm: false } );
-		}
-	}
 
-	getClickHandler = ( action, callback ) => {
-		return ( event ) => {
-			this.props.recordGoogleEvent( 'Me', 'Clicked on ' + action );
-
-			if ( callback ) {
-				callback( event );
-			}
-		};
-	};
-
-	handleApplicationNameFocus = () => {
-		this.props.recordGoogleEvent( 'Me', 'Focused on Application Name Field' );
-	};
+function ApplicationPasswords() {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
 
-	createApplicationPassword = ( event ) => {
-		event.preventDefault();
-		this.setState( { submittingForm: true } );
-		this.props.createApplicationPassword( this.state.applicationName );
-	};
+	const [ applicationName, setApplicationName ] = useState( '' );
+	const [ showAddPasswordForm, setShowAddPasswordForm ] = useState( false );
 
-	clearNewApplicationPassword = () => {
-		this.props.clearNewApplicationPassword();
-		this.setState( this.constructor.initialState );
-	};
+	const { data: appPasswords } = useAppPasswordsQuery();
+	const {
+		createApplicationPassword,
+		isLoading: isCreatingAppPassword,
+		data: newAppPasswordData,
+		reset: clearNewApplicationPassword,
+	} = useCreateAppPasswordMutation( {
+		onError() {
+			dispatch(
```

Comment:
I like the idea to keep the notices at a component level in this case. It might get noisy if this was used in multiple places, but since it has a single-use here, it makes sense for the component to handle and display the errors.